src/app/shared/address/tf-address.component.ts

```diff
@@ -0,0 +1,37 @@
+import { ContentfulConfig } from './../../contentful/tf-contentful.config';
+import { ContentfulService } from './../../contentful/tf-contentful.service';
+
+import { AddressModel } from './tf-address.model';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs/Subscription';
+
+@Component({
+  selector: 'tf-address',
+  templateUrl: 'tf-address.component.html'
+})
+
+export class AddressComponent implements OnInit, OnDestroy {
+  addressData: AddressModel;
+  private addressSubscription: Subscription;
```

Comment:
This is correct. But in our application going forward we will be using a different solution, I would like to see that implemented. Please follow this [stackoverflow](https://stackoverflow.com/a/41177163/6553077) discussion and make those changes. In case you need help let's have a discussion tomorrow.